R/tbl_filter.R

```diff
@@ -0,0 +1,146 @@
+#' Filter Hierarchical Tables
+#'
+#' @description `r lifecycle::badge('experimental')`\cr
+#'
+#' This function is used to filter hierarchical table rows. Filters are not applied to summary or overall rows.
+#'
+#' @param x (`tbl_hierarchical`, `tbl_hierarchical_count`)\cr
+#'   A hierarchical gtsummary table of class `'tbl_hierarchical'` or `'tbl_hierarchical_count'`.
+#' @param filter (`expression`)\cr
+#'   An expression that is used to filter rows of the table. See the Details section below.
+#' @param keep_empty (scalar `logical`)\cr
+#'   Logical argument indicating whether to retain summary rows corresponding to table hierarchy sections that have had
+#'   all rows filtered out. Default is `FALSE`.
+#' @inheritParams rlang::args_dots_empty
+#'
+#' @details
+#' The `filter` argument can be used to filter out rows of a table which do not meet the criteria provided as an
+#' expression. Rows can be filtered on the values of any of the possible statistics (`n`, `p`, and `N`) provided they
+#' are included at least once in the table, as well as the values of any `by` variables. Filtering is only applied to
+#' rows that correspond to the innermost variable in the hierarchy - all outer variable (summary) rows are kept
+#' regardless of whether they meet the filtering criteria themselves. In addition to filtering on individual statistic
+#' values, filters can be applied across the row (i.e. across all `by` variable values) by using aggregate functions
+#' such as `sum()` and `mean()`.
+#'
+#' If an overall column was added to the table (via `add_overall())`) this column will not be used in any filters (i.e.
+#' `sum(n)` will not include the overall `n` in a given row). To filter on overall statistics use the `sum()` function
+#' in your filter instead (i.e. `sum(n)` is equal to the overall column `n` across any `by` variables).
+#'
+#' Some examples of possible filters:
```

Comment:
Can you add a description of each of these filter explaining them like we did in the cards function?

Reply:
Done!